### PR TITLE
upgrade react-native 0.73.5

### PR DESCRIPTION
**Pull request by gabrieldonadel**
# Why

Follow up https://github.com/expo/expo/pull/26774 for react-native 0.73 upgrade 

# How

- update package versions
  - `react-native 0.73.4 -> 0.73.5`   
  
# Test Plan

- bare-expo ios / android
- Expo Go
- fabric ios / android
- ci passed
 

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by tsapeta — APPROVED** [review]

File: docs/public/static/diffs/template-bare-minimum/raw/45..50.diff [hunk 1]
```diff
@@ -2500,10 +2500,10 @@ index 69e7eb9619..4cc4f0aaf5 100644
 -    "react-dom": "17.0.2",
 -    "react-native": "0.68.2",
 -    "react-native-web": "0.17.7"
-+    "expo": "~50.0.6",
++    "expo": "~50.0.11",
 +    "expo-status-bar": "~1.11.1",
 +    "react": "18.2.0",
-+    "react-native": "0.73.2"
++    "react-native": "0.73.4"
```
Comment (brentvatne) [inline docs/public/static/diffs/template-bare-minimum/raw/45..50.diff:2506]: should this be 0.73.5?

File: docs/public/static/diffs/template-bare-minimum/raw/45..unversioned.diff [hunk 2]
```diff
@@ -2476,7 +2476,7 @@ index 69e7eb9619..150bb72764 100644
 +    "expo": "~50.0.0-preview.4",
 +    "expo-status-bar": "~1.11.0",
 +    "react": "18.2.0",
-+    "react-native": "0.73.2"
++    "react-native": "0.73.4"
```
Comment (brentvatne) [inline docs/public/static/diffs/template-bare-minimum/raw/45..unversioned.diff:2479]: ditto re: shouldn't this be 0.73.5?

Comment (brentvatne) [reply docs/public/static/diffs/template-bare-minimum/raw/45..unversioned.diff:2479]: (similar things in other diffs)

File: docs/public/static/diffs/template-bare-minimum/raw/45..50.diff (hunk 1 above)
Comment (gabrieldonadel) [reply docs/public/static/diffs/template-bare-minimum/raw/45..50.diff:2506]: Yeah, not sure why `et generate-bare-diffs` generated these changes

Comment (gabrieldonadel) [reply docs/public/static/diffs/template-bare-minimum/raw/45..50.diff:2506]: Here you go https://github.com/expo/expo/pull/27520

**Pull request merged**